src/main/java/Duke.java

```diff
@@ -1,10 +1,266 @@
+
+        printStart();
+        initialiseList();
+        Scanner scan = new Scanner(System.in);
+        while(true){
+            int num = 1;
+            //scan input
+            String input = scan.nextLine();
+            //check input and act based on different commands
+            try {
+                if (input.equals("bye")) {
+                    break;
+                } else if (input.startsWith("done ")) {
+                    Task task = makeDone(list, input);
+                    printDone(task);
+                    saveTask(list);
+                } else if (input.equals("list")) {
+                    printList(list);
+                } else if (input.startsWith("todo")) {
+                    Todo todo = createTodo(input,false);
+                    list.add(todo);
+                    todo.printTodo();
+                    saveTask(list);
+                } else if (input.startsWith("deadline")) {
+                    Deadline deadline = createDeadline(input,false);
+                    list.add(deadline);
+                    deadline.printDeadline();
+                    saveTask(list);
+                } else if (input.startsWith("event")) {
+                    Event event = createEvent(input,false);
+                    list.add(event);
+                    event.printEvent();
+                    saveTask(list);
+                }else if (input.startsWith("delete ")){
+                    Task task = makeDelete(list,input);
+                    printDelete(task,list.size());
+                    saveTask(list);
+                } else {
+                    throw new DukeException();
+                }
+            }catch (DukeTodoException e) {
+                printTodoError();
+            }catch (DukeDeadlineException e) {
+                printDeadlineError();
+            }catch (DukeEventException e) {
+                printEventError();
+            }catch (DukeException e) {
+                printError();
+            }catch (IOException e){
+                System.out.println("Writing to file failed.");
+            }
+        }
+        printBye();
+    }
+
+    private static void initialiseList() {
+        try {
+            getFile();
+        }catch(FileNotFoundException e){
+            try{
+                createFile();
+                System.out.println("File not found. Creating file...");
+            }catch(IOException e1){
+                System.out.println("Creating file failed.");
+            }
+        }
+    }
+
+    private static void getFile() throws FileNotFoundException {
+        Scanner scan = new Scanner(file);
+        while(scan.hasNextLine()){
+            String content = scan.nextLine();
+            String[] contents = content.split("\\s\\|\\s");
+            String legend = contents[0].trim();
+            Boolean done = Boolean.valueOf(contents[1].trim());
+            String action = contents[2].trim();
+            String action2 = "";
```

Comment:
The variable name "action2" is a bit confusing. For example, for Deadline, the "action2" should be the data for time the deadline task is due. Thus, isn't it better to name this variable something like "note" or "additionalInfo".